### PR TITLE
Port to the [dune] build system.

### PR DESCRIPTION
**Pull request by rlepigre**
This goes a bit further than #567, but there are still things to figure out:
- [ ] Making sure CI still works.
- [ ] Figure out whether there are requirements for Coq CI.
- [ ] Figure out how to turn more tests into cram tests (to get more silent build).
- [x] Figure out how to explicitly tell dune about the generated `coq-builtin[-synterp].elpi` files so that we can promote.
- [x] Fix doc target

### CODE REVIEW

File: apps/derive/elpi/dune [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  (glob_files *.elpi))
+ (action
+  (with-stdout-to %{target}
+   (echo ""))))
+
+(install
+ (files
+  (glob_files (*.elpi with_prefix coq/user-contrib/elpi/apps/derive/elpi/)))
+ (section lib_root)
+ (package coq-elpi))
```
Comment (rlepigre) [inline apps/derive/elpi/dune:18]: This setup seems to work pretty well, and takes care of installing `.elpi` files.

The only drawback is that we need to put these files under a different path (here files are found under `elpi.apps.derive.elpi`, while they were under `elpi.apps.derive` before). That's probably not a big deal?

File: builtin-doc/dune [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ (libraries elpi_plugin))
+
+(rule
+ (targets
+  coq-builtin.elpi
+  coq-builtin-synterp.elpi
+  elpi-builtin.elpi)
+ (deps gen_doc.exe)
+ (mode promote)
+ (action (run ./gen_doc.exe)))
```
Comment (rlepigre) [inline builtin-doc/dune:12]: With this setup, the `Elpi Document Builtins` is probably useless?

**Comment by gares** [comment]
I'll rebase and try to fix the doc target, which is a blocker

File: .nix/coq-overlays/coq-elpi/default.nix [hunk 3]
```diff
@@ -61,10 +61,9 @@ in mkCoqDerivation {
   release."1.6.0".sha256      = "0kf99i43mlf750fr7fric764mm495a53mg5kahnbp6zcjcxxrm0b";
   releaseRev = v: "v${v}";
 
-  buildFlags = [ "OCAMLWARN=" ];
-
   mlPlugin = true;
-  propagatedBuildInputs = [ coq.ocamlPackages.findlib elpi ];
+  useDune = true;
+  propagatedBuildInputs = [ coq.ocamlPackages.dune_3 elpi coq.ocamlPackages.ppx_optcomp ];
```
Comment (gares) [inline .nix/coq-overlays/coq-elpi/default.nix:66]:
```suggestion
  propagatedBuildInputs = [ coq.ocamlPackages.dune_3 coq.ocamlPackages.findlib  elpi coq.ocamlPackages.ppx_optcomp ];
```

**Pull request merged**